### ISSUE TITLE
Representing VAT component of a price

### ISSUE THREAD

**Issue by danbri** [schemaorg/schemaorg#380]
* There is currently no way to indicate the VAT portion of a PriceSpecification
* We do have a boolean http://schema.org/valueAddedTaxIncluded
* Most countries have a VAT system, http://en.wikipedia.org/wiki/Value-added_tax#Different_systems
* We should somehow make it possible to indicate how much of a price is from VAT

/cc @vholland

**Comment by sandstrom** [comment]
:clap:

**Comment by chaals** [comment]
+1 to Sandstrom

**Comment by danbri** [comment]
Rescheduled for Q2. @vholland do you have a sense how close we are?

**Comment by vholland** [comment]
I was thinking something along the lines of @sandstrom's suggestion: a vatAmount. I think I could get something together to propose with examples after sdo-gozer.

**Comment by danbri** [comment]
Thanks, that would be great. I'll leave it as Q2 for now. We'll need a codename for the next release soon btw...  how about sdo-tully? per http://ghostbusters.wikia.com/wiki/Louis_Tully QOTD ""Who does your taxes...?"

**Comment by danbri** [comment]
Ping - where are we with this?

**Comment by mfhepp** [comment]
We are almost one year older, wiser, and a few hairs have changed their colors to a grayish tone ;-)
Other than that, the issue is still pending...

**Comment by mfhepp** [comment]
But in any case, we need something that can handle relative price components, like a percentage on the unit price. A quick fix like vatAmount won't help IMO. Maybe a new subtype of PriceSpecification like TaxSpecification with properties like

taxPercentage
applicableRegion
exemptRegions

We will also need a mechanism to link the tax to the price components. This can either be done by adding a taxInformation property that links PriceSpecifications to TaxSpecifications or Number, or by a
priceReference property to link to the price components that are subject to the tax.

If we opt for the second route, we should encourage the use http://sdo-hotels.appspot.com/CompoundPriceSpecification in such scenarios.

**Comment by sandstrom** [comment]
Feel free to disagree with me, but would a relative tax percentage really be needed?

Mostly a `vatAmount` would be enough to back out the vat percentage if needed (` vat-amount / (total – vat-amount) = vat-rate`).

If the total price is a compound of multiple items (with different rates) this won't work. In those cases a per item price (with the vat amount) would work though. I assume it's already possible to specify prices both in aggregate and with specificity.

***
Regarding regions, for invoice/receipt type data that wouldn't make a difference. The amount is what it is, i.e. if you somehow get except you'll either have a `0` vat amount, or you'll somehow reclaim it later (in which case the vat amount is whatever is on the invoice/receipt).

**Comment by sandstrom** [comment]
@danbri @vholland what's the state on this? This would be an awesome enhancement! ⛵️

**Comment by Download** [comment]
I found this issue looking for a way to include VAT information in the price specification.

It seems there is no way?

Would it be an idea to set `valueAddedTaxIncluded` on the `CompoundPriceSpecificiation` to `true` to indicate it includes VAT, then have the rule that there must be exactly one price component that also has this flag set to `true` to indicate that that component is the VAT part of the total price? As @sandstrom correctly remarks above, if you have the grand total and all components, calculating the VAT percentage is easy. If you need this kind of detail information on the item level, just include compound price specs for all items. Otherwise you can leave it to the top level.

**Example:**

```json
{
  "@context": "https://schema.org",
  "@type": "Product",
  "name": "Implement VAT for Structured Data",
  "offers": {
    "@type": "Offer",
    "priceSpecification": {
      "@type": "CompoundPriceSpecification",
      "name": "Price",
      "price": "121.00",
      "priceCurrency": "EUR",
      "valueAddedTaxIncluded": true,
      "priceComponent": [
        {
          "name": "Price excluding VAT",
          "price": "100.00",
          "valueAddedTaxIncluded": false
        },
        {
          "name": "VAT",
          "price": "21.00",
          "valueAddedTaxIncluded": true
        }
      ]
    }
  }
}
```

Simply creating the rule that:

* If a `CompoundPriceSpecification` has `valueAddedTaxIncluded` set to `true`, **AND**
* It includes exactly **ONE** `UnitPriceSpecification` in its `priceComponent` that also has `valueAddedTaxIncluded` set to `true`, **THEN**
* The `UnitPriceSpecification` with the `valueAddedTaxIncluded` set to `true` contains the VAT amount

would allow us to add VAT information in a structured way without having to change any schema. In fact just writing this down in the docs would probably be enough to create enough clarity to make this workable. Parsing should already work and if agents include this rule they will be able to find VAT amounts for prices easily. 

It's been over 2 years since this issue has been opened. VAT is an important part of e-commerce so it is pretty strange imho that we still have no good solution. It should be much higher priority I think.

In the meantime any advice on workarounds would be very useful. Would the above be a good workaround or does it go against (the spirit of) the current schema?

**Comment by Download** [comment]
@RichardWallis Any thoughts? I am asking you because I noticed you have 'issue closing powers'...

**Comment by RichardWallis** [comment]
@Download 
> Simply creating the rule...

That is something that doesn't apply in Schema.org - we can suggest and provide examples, but there are [currently] no ways to enforce such rules.

Back to the issue in question...

Looking at [PriceSpecification](http://schema.org/CompoundPriceSpecification), which already contains the _valueAddedTaxIncluded_ property, would adding a _valueAddedTaxAmount_  property be a pragmatic solution.

If the _price_ property is used, the use of _valueAddedTaxIncluded_  and _valueAddedTaxAmount_  would make sense.  It however would be ambiguous if maxPrice & minPrice are used.

So:
``` {
  "@context": "https://schema.org",
  "@type": "Product",
  "name": "Implement VAT for Structured Data",
  "offers": {
    "@type": "Offer",
    "priceSpecification": {
      "@type": "PriceSpecification",
      "name": "Price",
      "price": "121.00",
      "priceCurrency": "EUR",
      "valueAddedTaxIncluded": true,
      "valueAddedTaxAmount": "21.00"
    }
  }
}
```
or 
```
    "priceSpecification": {
      "@type": "PriceSpecification",
      "name": "Price",
      "price": "100.00",
      "priceCurrency": "EUR",
      "valueAddedTaxIncluded": false,
      "valueAddedTaxAmount": "21.00"
    }
```

**Comment by sandstrom** [comment]
@RichardWallis Yes, that would do it (it could also be named `valueAddedTax`, but both names work).

Also, `valueAddedTaxIncluded` should be deprecated (the presence of `valueAddedTaxAmount` will itself signify the same thing as the boolean).

**Comment by RichardWallis** [comment]
> Also, valueAddedTaxIncluded should be deprecated 

You can only do that if there is a rule that a price always includes the VAT amount if stated - (see my above comment about rules)

**Comment by Download** [comment]
> > Simply creating the rule...
> That is something that doesn't apply in Schema.org - we can suggest and provide examples, but there are [currently] no ways to enforce such rules.

Yes I get that. But since atm there is no real guidance at all, simply stating that 'this is the way we envision it should be used' would be pretty close to a rule. Not all rules need to be enforced to have value.

I think your example is a lot more concise than mine. That's a big plus! The downside is it introduces the concept of multiple prices in a single price specification, bypassing CompoundPriceSpecification. I envision more complex compound price specifications including discounts, fees etc. Value Added Tax would be just one of the many components of the total price. Like we see on invoices.

I had to make a decision for my client (deadline) so I went with what I proposed. But either way would be ok I think, as long as there is some clear guidance from schema.org so people can look up how it's supposed to work. In the end people will just be copy-pasting your examples and tweaking it. So if we set a good example, it will be followed.

Please don't deprecate anything. It's not worth the pain. There is no harm in an extra property. 

/rant on
I've seen the idea to deprecate stuff being proposed here on various issues. But please heed the lesson that W3C learned. This is the web. Deprecating stuff is nigh impossible. And not very productive either. All it ends up doing is invalidating stuff that will be around for years to come. Instead, try to document existing practices and add only the minimum of stuff needed to get things done that can't be done easily without a standards body being involved. W3C tried to deprecate `<b>` and `<i>`. After years of warnings/blog posts/flame wars about that stuff being deprecated it just came back in HTML5 because deprecating it had simply failed. If only we had spend that time on doing more productive things. The same with enforcing rules and schema validation etc. It just won't work. People will be copy-pasting code snippets with the deprecated stuff in it for years to come. Let's just add the minimum of stuff needed to get VAT to work in a sane way.
/rant off

**Comment by Download** [comment]
I ended up implementing my proposal. It is now in active use on a few dozen high profile eyewear related websites.

**Comment by wuda-io** [comment]
Hello Folks,

i also used the way from @Download 

Why:
This looks promising, because I tried some other things with "vatAmount" before. I also thought this would be nice but at some items there are actually more than one Tax... for example at Cars there is additionally to the VAT also the so called NoVA which is a special Tax in some EU Countries.
With this in mind, i would approve this way because it also allows multiple VATs per Product/Service Item.

Thanks for reading .... happy weekend

**Comment by ldexterldesign** [comment]
I don't apologise for necroing 🧟 this because I seem to come back to it year on year and it needs a solution

FYI I'm in UK (unfortunately, right now)

My use case is I buy (the same) products online and offline and I want granular info about what things cost with and without VAT so I can compare them later:
* When I buy online there's always a product price, shipping price and total price; and the VAT is typically excluded until checkout when it's applied and - if I'm lucky - then displayed itemised with and without VAT so I can make sense of prices
* When I buy offline (i.e. in a retail food shop) the VAT is typically included (if I am in a wholesalers then it's typically excluded)

I'm going to have to go with @Download's [solution] 👏 here because I haven't read anything better over these issues and I think it should get implemented in https://schema.org

--

I think '[CompoundPriceSpecification.priceType]' should be utilised within the [solution] component structure to indicate [VAT nuances] and in [Offers] to indicate [discounts]

7 years is long enough to not have something in the spec' for this

Sincerely

[discounts]: https://github.com/schemaorg/schemaorg/issues/1742
[solution]: https://github.com/schemaorg/schemaorg/issues/380#issuecomment-313699748
[CompoundPriceSpecification]: https://schema.org/CompoundPriceSpecification
[Offers]: https://github.com/schemaorg/schemaorg/issues/2677
[VAT nuances]: https://github.com/openactive/open-booking-api/issues/77

**Comment by fthobe** [comment]
@danbri is this still a topic?

**Comment by fthobe** [comment]
Hey @Download , there's a discussion about reviving this issue. I would like to take a look at it if that's ok for you!

Can you send me your implementation?

**Comment by fthobe** [comment]
@alex-jansen @rrlevering 

ok, I had some thought about this: 
we should add following properties to Price: 

- [ ] `priceExcludingTaxes` (number) to indicate the price without taxes to validate priceTax and price against (optional);
- [ ] `priceTax` (number) to indicate the actual tax amount included;
- [ ] `priceTaxRate` (number) to indicate the actual tax rate applied;
- [ ] `businessTaxExemption` (boolean) to describe if a B2B customer can purchase without Sales Taxes (exists in Europe, US and many other jurisdictions). 

This allows to create features to filter: 
- For B2B offerings as businesses frequently do not pay those taxes and have a hard time to identify stores that can remove taxes from the final price (see example of what Amazon does); 
- Show net prices or toggle between both prices; 
- Show visualize vat rate as well as total tax. 

Why I am not in favour of adding fees such as consumer electronics recycling fees: 
- they are not deductable therefor the information does not add value;
- they are often not modelled in the ecommerce system. 

Why I am not in favour of adding customs?
- more merchants than less will get it wrong; 
- custom codes are hard to deal with. 

# Example Visualization of Taxes for Business Users on Amazon
<img width="1223" alt="Image" src="https://github.com/user-attachments/assets/32971b46-7817-4fb1-bf30-142510dd9d71" />

**Comment by fthobe** [comment]
@danbri hardly any eCommerce moves outside of an economic zone (eg outside of US or EU). 

Most merchants do not offer it. 

To break it down: 
- B2C is almost irrelevant for export outside of the economic zone in EU
- B2B is relevant but customs get usually handled by brokers, I do not believe there is an additional value (but lot's of headache) related to it, I know UK is a special case here, but i would like to sort that out separately

**Comment by fthobe** [comment]
Hey @alex-jansen Do you have the time to take a look?

**Comment by alex-jansen** [comment]
After reading this thread there seems to be some consensus about @Download's [proposal](https://github.com/schemaorg/schemaorg/issues/380#issuecomment-313699748) leveraging CompoundPriceSpecification. It also seems consistent with #830 and #2689 to allow modeling of prices with multiple price components, such as for car rentals, hotels, and mobile phones. We could decide to add a new enum "https://schema.org/VAT" to https://schema.org/PriceComponentTypeEnumeration used as range for https://schema.org/priceComponentType, to allow for a specific VAT price component part of a total price. We could then use that in combination with the existing http://schema.org/valueAddedTaxIncluded. Using the generic CompoundPriceSpecification also allows future modeling of recycling and other fees. Thoughts?

**Comment by fthobe** [comment]
Hey, I think that in General price component would be a good way to go and that it is a great idea.

Some notes though on implementation: 
The general idea is that for companies VAT is not a cost, but only a cashflow position than often can also be removed from the purchase price (eg Reverse Charge in Europe or Sales Tax Exemption in the US).  
- the current offer / price model seems to me very battle tested, are you sure incrementalising prices into a more complex schema that contains sub properties wouldn't hinder adoption? Is there widespread adoption of componentPrice (out of curiosity)? 
- is the vat rate (eg 22%) something that is considered relevant? It is on some systems hard to pull. 
- how incremental do you want the components (only VAT or also other fees and taxes)? I think an additional taxes position is inevitable.
- I saw that the enum currently mixes billable services such as [DistanceFee](https://schema.org/DistanceFee) and payments  such as [Downpayment](https://schema.org/Downpayment), do you feel that's ok or confusing?

Technical Questions:
- there should be a toggle for sales tax exemptions support, the EU is pushing to make reverse charge the standard and any schema should support indicating if a business does offer invoices without VAT; 
- I am struggling a little with @Download 's implementation of `valueAddedTaxIncluded` as I think he mixed up some things there;
- what do you want to do when a price component is itself a price with multiple vat rates?

Example of Price Components of an Airbnb listing: 

Room Price 
 | - Cleaning Fee (regular service vat) 
 | - City Tax (No vat as passing through taxes from tourist to the city)

Smartphone
|- Net Price 
|- Other taxes (IP tax in Germany for example)
|- VAT

I'd consider picking up the @download proposal bringing it in shape, but I feel we are running to fast as there's a lot to consider here.

**Comment by fthobe** [comment]
@Download can you give me an example of an implementation? What was the reasoning behind it?

**Comment by fthobe** [comment]
@wuda-io 
> That or even better would be to use general Tax (https://schema.org/Tax) because there can be Taxes which does not have to be a VAT explicilty. But otherwise fully agree.

I think we should allow both: 
- add non deductable taxes to net or add them 

or 
 
- outline them individually. 

What's your thought on that?

**Comment by fthobe** [comment]
@alex-jansen do you think there should be an option for shipping cost as well inside the components? I know you people have done a lot of great work there recently and I don't want to interfere or waste anybody's time on stuff without chances for implementation?

**Comment by fthobe** [comment]
@alex-jansen I have a technical question: would the extended enum vocabulary go into the [issue-2689.ttl](https://github.com/schemaorg/schemaorg/blob/54ecf1171c51283c35ba5da0bbfd2e7de497f27e/data/ext/pending/issue-2689.ttl#L14) or do I need to add a new file to pending?

**Comment by alex-jansen** [comment]
Re. Shipping Cost: I would not add shipping cost under price components since shipping cost modeling is an entirely separate complex modeling exercise, with many lookup tables and regional-dependencies. However, one difficulty I see is that usually a merchant has to charge VAT on shipping costs, even for domestic shipments, so that begs the question whether you want to expand the shipping model to include VAT as well.

In terms of the vocabulary I would add new enum values for an existing enum type where they were originally defined to keep things together as much as possible.

**Comment by fthobe** [comment]
> However, one difficulty I see is that usually a merchant has to charge VAT on shipping costs, even for domestic shipments, so that begs the question whether you want to expand the shipping model to include VAT as well.

Depends on the country and sales tax exemptions support also there

**Comment by Download** [comment]
> [@Download](https://github.com/Download) can you give me an example of an implementation? What was the reasoning behind it to implement VAT without consuming systems?

Sorry for the late response. This issue took some time at the schema.org side, and we had to pull the trigger so I ended up implementing my proposal and just using it.

In our case it was a decision to try to integrate the data we (websites we are servicing) publish for search engines (schema.org format) with the data we publish for analytics software (all have custom schemes) and then make an adapter layer that could translate 'standardized' schema.org metadata to the format expected by the analytics layer we want to send it to.

I ended up regretting that decision as I concluded schema.org was too immature for it. The potential is there. There is like a 95 percent overlap. In particular, this issue was about the fact that the analytics software wants the VAT amounts separately so it can show sales reports consistently and not 'inflate' sales by 21 percent (Dutch VAT rate) or the opposite. The websites that this software was used on would publish their data like prices etc as part of the normal web page (in the schema data). Every shop has VAT sorted out, there just was no clear way to include VAT in a way that made it easy to extract it again later... Which is what my proposal was trying to add.

A non-perfect solution is often much better than no solution at all...

**Comment by Download** [comment]
Also, after having been away from schema.org for a while (and leaving highly frustrated, to be honest), coming back here feels like 'oh, yeah, this is why I left here and never looked back. Sorry to rant but you guys have a SERIOUS issue, ok?

This issue has been open for 10 YEARS. Program the bot to mark it as LONG OVERDUE or something i.s.o. letting it mark it as 'stale'...

WHY is that taking so long? I'll tell you why:

1. Too many discussion. Too much talking, not enough fixing
2. The Utopia perspective; if we are fixing this one thing, then lets also fix everything until it is perfect...
3. Feature creep; Like 2., but if we add this, we should also add (100 things)...
4. This proposal is good, but (a thousand things that could be changed/improved)
5. Discussing the 1000 things from 4....
6. Closing the issue every few weeks as 'inactive; as soon as discussion dies down
7. Forcing people to discuss it more, just to keep the issue open
8. Discussing why it is closed...

### My proposal:

* Do NOT change the format/specification/structure of the markup
* Do NOT add new properies/enums/etc
* USE what is there
* Document it!!
* Provide some examples in the right places (these will be copied, over and over, they will set the standard. You just watch and see)
* Yes, you guys DO have the POWER. No need to wait for W3C or Google or anyone. This spec is YOURS. If Google does not like what you changed, just open a new issue and fix it. Quickly...
* One issue at a time!!
* Don't close issues unless they are resolved.
* Create new issues for all the spin-off ideas

There is more. But you guys need to take your power and use it. Just decide and move forward. You can do it!

@fthobe I am willing to discuss whatever you want to discuss. Lets find a way to talk 👍
